base/strings/basic.jl

```diff
@@ -138,7 +138,7 @@ function nextind(s::AbstractString, i::Integer)
         return 1
     end
     if i > e
-        return i+1
+        return sizeof(s)+1
```

Comment:
This doesn't seem right, because this `nextind` method is supposed to be for `AbstractString`, and `sizeof` may not be the right thing for a generic string-like object.